packages/dev/core/src/Misc/tools.ts

```diff
@@ -45,6 +45,39 @@ export class Tools {
         FileToolsOptions.BaseUrl = value;
     }
 
+    /**
+     * This function checks whether a URL is absolute or not.
+     * It will also detect data and blob URLs
+     * @param url the url to check
+     * @returns is the url absolute or relative
+     */
+    public static IsAbsoluteUrl(url: string): boolean {

```

Comment:
We should probably also fix GetAbsoluteUrl to use this function instead of erroring out when the passed in argument is already absolute.

Reply:
can do! i didn't want to touch this function, but i can add the check at the beginning.

Reply:
We can do it separately. It probably better to do it only in the final else case.

Reply:
Let's do it separately, as I want to understand in what cases this will fail